android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BeaconConfigFragment.java

```diff
@@ -172,6 +174,17 @@ public void onPause() {
   }
 
   @Override
+  public void onDestroy() {
+    super.onDestroy();
+    if (mShortenUrlTask != null) {
```

Comment:
I think this one needs to me moved to onPause() since the connection to the beacon gets closed there.